extensions/query-store/src/reports/baseQueryStoreReport.ts

```diff
@@ -0,0 +1,145 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as azdata from 'azdata';
+import * as path from 'path';
+import * as utils from '../common/utils';
+import * as constants from '../common/constants';
+
+export abstract class BaseQueryStoreReport {
+	protected editor: azdata.workspace.ModelViewEditor;
+	protected flexModel?: azdata.FlexContainer;
+
+	constructor(reportName: string, private reportTitle: string, protected resizeable: boolean, private extensionContext: vscode.ExtensionContext) {
+		this.editor = azdata.workspace.createModelViewEditor(reportName, { retainContextWhenHidden: true, supportsSave: false }, reportName);
+	}
+
+	/**
+	 * Creates and opens the report
+	 */
+	public async open(): Promise<void> {
+		this.editor.registerContent(async (view) => {
+			this.flexModel = <azdata.FlexContainer>view.modelBuilder.flexContainer().component();
+
+			const toolbar = await this.createToolbar(view);
+			this.flexModel.addItem(toolbar, { flex: 'none' });
+
+			const views = await this.createViews(view);
+
+			const mainContainer = await this.createMainContainer(view, views);
+
+			this.flexModel.addItem(mainContainer, { CSSStyles: { 'width': '100%', 'height': '100%' } });
+
+			this.flexModel.setLayout({
+				flexFlow: 'column',
+				height: '100%'
+			});
+
+			await view.initializeModel(this.flexModel);
+		});
+
+		await this.editor.openEditor();
+	}
+
+	/**
+	 * Creates the main container containing the different components of the report
+	 * @param view
+	 * @param containers Array of containers to add to the main container
+	 * @returns FlexContainer or SplitViewContainer containing the containers
+	 */
+	private async createMainContainer(view: azdata.ModelView, containers: azdata.FlexContainer[]): Promise<azdata.FlexContainer | azdata.SplitViewContainer> {
+		let mainContainer;
+
+		switch (containers.length) {
+			case 1: {
+				mainContainer = view;
+			}
```

Comment:
Did you mean to omit the `break`? It's currently going to fall through to the next case without it.

Reply:
whoops good catch! I added this case later after the others and missed it